dspace-server-webapp/src/test/java/org/dspace/app/rest/VersionHistoryRestRepositoryIT.java

```diff
@@ -115,48 +167,452 @@ public void findVersionsOneWrongIDTest() throws Exception {
     }
 
     @Test
-    public void findVersionsOfVersionHistoryTest() throws Exception {
-        Version version = versionHistoryService.getFirstVersion(context, versionHistory);
-        getClient().perform(get("/api/versioning/versionhistories/" + versionHistory.getID() + "/versions"))
-                   .andExpect(status().isOk())
-                   .andExpect(jsonPath("$._embedded.versions", contains(VersionMatcher.matchEntry(version))));
+    public void findVersionsOfVersionHistoryAdminTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                                          .withName("Parent Community")
+                                          .build();
+
+        Collection col = CollectionBuilder.createCollection(context, parentCommunity)
+                                          .withName("Collection test")
+                                          .build();
+
+        Item item = ItemBuilder.createItem(context, col)
+                               .withTitle("Public test item")
+                               .withIssueDate("2021-04-27")
+                               .withAuthor("Doe, John")
+                               .withSubject("ExtraEntry")
+                               .build();
 
+        Version version = VersionBuilder.createVersion(context, item, "test").build();
+        VersionHistory versionHistory = versionHistoryService.findByItem(context, item);
+        Version version2 = versioningService.getVersion(context, item);
         context.turnOffAuthorisationSystem();
-        Version secondVersion = versioningService
-            .createNewVersion(context, versionHistory, item, "test", new Date(), 0);
+
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+        getClient(tokenAdmin).perform(get("/api/versioning/versionhistories/" + versionHistory.getID() + "/versions"))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$._embedded.versions", containsInAnyOrder(
+                                                 VersionMatcher.matchEntry(version),
+                                                 VersionMatcher.matchEntry(version2)
+                                                 )));
+
+        context.turnOffAuthorisationSystem();
+        Version version3 = VersionBuilder.createVersion(context, item, "test3").build();
         context.restoreAuthSystemState();
 
+        getClient(tokenAdmin).perform(get("/api/versioning/versionhistories/" + versionHistory.getID() + "/versions"))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$._embedded.versions", containsInAnyOrder(
```

Comment:
Shouldn't we be checking that these versions are in a very *specific* order instead of "AnyOrder"?  I believe versions are ordered by ID.  So, this should likely be `Matchers.containsInRelativeOrder(`